### PR TITLE
New toolchain step inside the validation step

### PR DESCRIPTION
**Pull request by ziopio**
I think this is an acceptable way to add a check for the tool-chain.
For now i limited the action by just checking the existence of the directory.

If the dir exists a message is printed otherwise an error is produced.

### CODE REVIEW

**Comment by ziopio** [comment]
[Detect missing/misconfigured toolchain earlier](https://trello.com/c/hBG43Kwc/10-detect-missing-misconfigured-toolchain-earlier)

**Comment by eproxus** [comment]
Checking the folder is a good start. We should also do smoke test check for some files, e.g. `bin/arm-rtems5-gcc` or `GRiSP_TOOLCHAIN_REVISION` (in an earlier step than we check today).

**Comment by ziopio** [comment]
I wrote this, so it's easy to add more elements to check.
This method does not check whether it be a directory or a file.

File: src/grisp_tools_build.erl [hunk 1]
```diff
@@ -26,7 +26,8 @@ run(Configuration) ->
         {fun build/1, [
             {validate, [
                 fun grisp_tools_step:apps/1,
-                fun grisp_tools_step:version/1
+                fun grisp_tools_step:version/1,
+                fun grisp_tools_step:toolchain/1
```
Comment (eproxus) [inline src/grisp_tools_build.erl:30]: I think the name `validate_toolchain` is more descriptive. Do you mind changing it?

File: src/grisp_tools_step.erl [hunk 2]
```diff
@@ -112,6 +113,21 @@ collect(#{platform := Platform} = S0) ->
     S2 = event(S1, [{hash, Hash, HashIndex}]),
     mapz:deep_put([build, hash], #{value => Hash, index => HashIndex}, S2).
 
+toolchain(S0) ->
+    ToolchainRoot = mapz:deep_get([paths, toolchain], S0),
+    [error({toolchain_root_invalid, ToolchainRoot}) || not filelib:is_dir(ToolchainRoot)],
+    FILES = [
```
Comment (eproxus) [inline src/grisp_tools_step.erl]: Standard in Erlang is to use camel case variable names. Can you change this to `Files` and `File` below?

**Review by eproxus — CHANGES_REQUESTED** [review]

File: src/grisp_tools_build.erl (hunk 1 above)
Comment (ziopio) [reply src/grisp_tools_build.erl:30]: ok

File: src/grisp_tools_step.erl (hunk 2 above)
Comment (ziopio) [reply src/grisp_tools_step.erl]: Mmm yep, I forgot this one, i fix

**Pull request merged**